src/event/mod.rs

```diff
@@ -130,6 +130,18 @@ impl LogEvent {
         self.fields.insert(key.into(), value.into());
     }
 
+    pub fn try_insert<V>(&mut self, key: &Atom, value: V) -> Option<V>
+    where
+        V: Into<Value>,
+    {
+        if self.contains(key) {
+            Some(value)
+        } else {
+            self.insert(key.clone(), value);
+            None
+        }
+    }
```

Comment:
I don't see any of the callers of this method using the return value. If true, I think it would be better to drop both result values until they are needed.

Also, would using the `Entry` interface make sense? That is: ```self.entry(key.clone()).or_insert(value);``` I don't know what is more efficient here.

Reply:
True, and regarding `Entry` we can't use `HashMap`'s `Entry` since the key can be composed of multiple parts so we need to go through `util::log::` methods.

Reply:
Right, of course.